### PR TITLE
[GPU] Bugfix mis-match axes size in ConvertMatMulToFullyConnected

### PR DESCRIPTION
**Pull request by byungilm**
+ Resolved execution failure of segnext model

### Details:
 - *item1*
 - *...*

### Tickets:
 - 146597

### CODE REVIEW

File: src/plugins/intel_gpu/src/plugin/transformations/convert_matmul_to_fc.cpp [hunk 1]
```diff
@@ -41,10 +41,21 @@ ConvertMatMulToFullyConnected::ConvertMatMulToFullyConnected() {
...
+        for (auto& user : input_b->get_users()) {
+            if (user != matmul && ov::is_type<ov::op::v0::MatMul>(user)) {
+                auto other_matmul = std::dynamic_pointer_cast<ov::op::v0::MatMul>(user);
+                if (input_b == other_matmul->get_input_node_shared_ptr(0))
+                    return false;
+            }
+        }
+
+        // fc_input_a and fc_input_b - are the final inputs that will be set to FullyConnected.
+        // So in case of adding new operations that takes matmul inputs we need keep update fc_input_a and fc_input_b.
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/plugin/transformations/convert_matmul_to_fc.cpp:58]: Could you please implement a test case?

I think we need to have more test to understand the impact of this change.
For example, input is connected to input1 of two matmuls(A, B) and transpose configuration may be different from A and B.

File: src/plugins/intel_gpu/src/plugin/transformations/convert_matmul_to_fc.cpp [hunk 2]
```diff
@@ -41,10 +41,21 @@ ConvertMatMulToFullyConnected::ConvertMatMulToFullyConnected() {
...
+        for (auto& user : input_b->get_users()) {
+            if (user != matmul && ov::is_type<ov::op::v0::MatMul>(user) && ov::is_type<ov::op::v0::Convert>(input_b)) {
+                auto other_matmul = std::dynamic_pointer_cast<ov::op::v0::MatMul>(user);
+                if (input_b == other_matmul->get_input_node_shared_ptr(0) || fc_input_b == fc_input_a)
+                    return false;
+            }
+        }
+
+        // fc_input_a and fc_input_b - are the final inputs that will be set to FullyConnected.
+        // So in case of adding new operations that takes matmul inputs we need keep update fc_input_a and fc_input_b.
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/plugin/transformations/convert_matmul_to_fc.cpp:58]: could you please review functionality and test coverage as discussed offline?

Comment (byungilm) [reply src/plugins/intel_gpu/src/plugin/transformations/convert_matmul_to_fc.cpp:58]: Applied

File: src/plugins/intel_gpu/src/plugin/transformations/convert_matmul_to_fc.cpp [hunk 3]
```diff
@@ -41,10 +41,24 @@ ConvertMatMulToFullyConnected::ConvertMatMulToFullyConnected() {
             return false;
         }
 
-        // fc_input_a and fc_input_b - are the final inputs that will be set to FullyConnected.
-        // So in case of adding new operations that takes matmul inputs we need keep update fc_input_a and fc_input_b.
         auto fc_input_a = pattern_map.at(activations_m);
         auto fc_input_b = pattern_map.at(weights_m);
+
+        // Not to convert fc_input_b which should be fc_input_a of other sibling MatMul
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/plugin/transformations/convert_matmul_to_fc.cpp:47]: nit:
suggestion for comment
```
// If fc_input_b is shared with another matmul, transposing fc_input_b is limited.
// If it is connected to input_a of another matmul, do not transpose
// If it is connected to input_b of another matmul and the transpose option is different between those two matmuls, do not transpose.
```

**Review by isanghao — APPROVED** [review]
LGTM. Minor comment was left.

File: src/plugins/intel_gpu/src/plugin/transformations/convert_matmul_to_fc.cpp (hunk 3 above)
Comment (byungilm) [reply src/plugins/intel_gpu/src/plugin/transformations/convert_matmul_to_fc.cpp:47]: Applied by other PR

**Pull request merged**